content/articles/binance-smart-chain-tutorial/index.md

```diff
@@ -0,0 +1,241 @@
+
+- Finally, our function emits a `Transfer` event to support transfer.
+
+### Approve and transfer
+To support delegate transactions, we need to approve the actions of the spender so that the owner of the account allows the delegate to transfer the requested amount of tokens. The `approve` function will limit the value of tokens that are deducted from the sender's balance. The code snippet for the `approve` function looks like this:
+
+```js
+// Delegate a Transfer Functionn
+function approve(address spender, uint value) public returns(bool){
+    allowance[msg.sender][spender] = value; // spender can spend *value* amount belonging to sender 
+    emit Approval(msg.sender, spender, value); // emit approval event to allow spending
+    return true;
+}      
+```  
+
+To transfer tokens from an account, we need to call `Approve` on the address of the `spender` within the smart contract itself.
+
+- The line `allowance[msg.sender][spender] = value;`: 
+
+- `emit Approval(msg.sender, spender, value);` will emit an event from `msg.sender` to approve the `spender` to spend 
+  `value` amount of tokens to its address within the smart contract.
+
+Lastly, add a function to automate transfers of approved delegates. The `transferFrom` function will perform a deduction from an account wallet of the owner and emit a `Transfer` event. Check the code below: 
+
+```js
+// 
+function transferFrom(address from, address to, uint value) public returns(bool){
+    // check allowance mapping if spender is approved
+    require(allowance[from][msg.sender] >=value, "allowance too low");  
+    // check balance
+    require(balanceOf(from) >= value, "balance is too low"); 
+    // update mappings balances of sender & recipient 
+    balances[to] += value;
+    balances[from] -= value;
+    // emit event
+    emit Transfer(from, to, value);
+    return true;
+}
+```  
+
+- The first line `require(allowance[from][msg.sender] >=value, "allowance too low");` retrieves a mapping of how much (amount `value` in tokens) the owner allows another account (the delegate) to transfer on his behalf. 
+
+- The `balanceOf` is a function that returns the amount in tokens the address has to confirm spending. Since we need to ensure they exist before allowing them to spend to restrict double spending
+
+- Finally, we update our mappings and emit the `Transfer` event to complete the transaction.
+
+### A final demo in MetaMask
+
+Using a metamask plugin, we can inject the web3.js namespace within our browser. If we configure and deploy our smart contract in a sandbox test environment, the final app should look like this:
+
+The executable functions that we deploy to the smart contract to run transactions.
+
+![executable functions](/engineering-education/binance-smart-chain-tutorial/execFuncs.png).
```

Comment:
Please make sure you name your files in small letters, i.e., execfuncs.png or exec-funcs.png